### PR TITLE
Implemented image list gathering based on labels

### LINKED ISSUE #16: Change the implementation of how image list is gathered to use labels
cc @mroman-redhat 

Do this after #15 

Currently the list of images is generated by getting all `ImageStreams` from the namespaces and filtering by their name - looking for suffix `-notebook`.

To follow the best practices, we should use `labels` (`metadata.labels`) to find the `ImageStream`s which we want to list.

If the `ImageStream` has a label `opendatahub.io/notebook-image: true`, it should be added to the list, ignored otherwise

FYI @erikerlandson

### PR DESCRIPTION
**Pull request by mroman-redhat**
Closes #16

### CODE REVIEW

**Comment by vpavlin** [comment]
Please rebase

File: jupyterhub_singleuser_profiles/images.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                selected = "selected=selected"
+            result.append("<option value='%s' %s>%s</option>" % (image_tag, selected, image_tag))
+
+
+    def get_form(self, last_image=None):
+
+        result = []
+        imagestream_list = self.get_images(IMAGE_LABEL+'=true')
+
+        if len(imagestream_list.items) == 0:
```
Comment (vpavlin) [inline jupyterhub_singleuser_profiles/images.py]: If this is evaluated as `False` the loop on the line 53 will fail

**Comment by crobby** [comment]
lgtm, but are there some specific steps I'd need to try this out?

**Comment by mroman-redhat** [comment]
I tested it by running odh-operator on openshift, and setting the source in jupyterhub-img build config to `https://github.com/mroman-redhat/jupyterhub-ocp-oauth` branch `issue-16`, this should make it use my commit of jupyterhub singleuser profiles. You can then start rollout of the deployment config and the jupyterhub notebook should start without problems. This doesn't cover all possibilities, such as differently named imagestreams, so @vpavlin is probably better at explaining how to test this. Sorry if I am vague.

**Comment by vpavlin** [comment]
LGTM
Screenshot from 2020-03-13 16-42-52

**Pull request merged**